crates/transpiler/src/passes/high_level_synthesis.rs

```diff
@@ -881,11 +937,15 @@ fn synthesize_op_using_plugins(
         OperationRef::Operation(operation) => operation.operation.clone_ref(py),
         OperationRef::Unitary(unitary) => unitary.create_py_op(py, label)?.into_any(),
     };
-
     let res = HLS_SYNTHESIZE_OP_USING_PLUGINS
         .get_bound(py)
-        .call1((op_py, input_qubits, data, tracker.clone()))?
-        .extract::<Option<(QuantumCircuitData, Vec<usize>)>>()?;
+        .call1((
+            op_py,
+            input_qubits.iter().map(|x| x.0).collect::<Vec<_>>(),
```

Comment:
This intermediate collection would also be unnecessary if we added an `impl IntoPyObject` for `Qubit`.